src/app/views/communities/view/overview/edit/games/games.ts

```diff
@@ -0,0 +1,139 @@
+import { Emit } from 'vue-property-decorator';
+import { arrayRemove } from '../../../../../../../utils/array';
+import { Api } from '../../../../../../../_common/api/api.service';
+import AppCardListAdd from '../../../../../../../_common/card/list/add/add.vue';
+import AppCardListDraggable from '../../../../../../../_common/card/list/draggable/draggable.vue';
+import AppCardListItem from '../../../../../../../_common/card/list/item/item.vue';
+import AppCardList from '../../../../../../../_common/card/list/list.vue';
+import { Game } from '../../../../../../../_common/game/game.model';
+import AppGameThumbnailImg from '../../../../../../../_common/game/thumbnail-img/thumbnail-img.vue';
+import { Growls } from '../../../../../../../_common/growls/growls.service';
+import {
+	BaseRouteComponent,
+	RouteResolver,
+} from '../../../../../../../_common/route/route-component';
+import { AppTooltip } from '../../../../../../../_common/tooltip/tooltip';
+import { CommunityLinkGameModal } from '../../../../../../components/community/link-game-modal/link-game-modal.service';
+import { AppCommunityPerms } from '../../../../../../components/community/perms/perms';
+import { RouteStore, RouteStoreModule } from '../edit.store';
+
+@Component({
+	name: 'RouteCommunitiesViewEditGames',
+	components: {
+		AppCommunityPerms,
+		AppCardList,
+		AppCardListDraggable,
+		AppCardListItem,
+		AppCardListAdd,
+		AppGameThumbnailImg,
+	},
+	directives: {
+		AppTooltip,
+	},
+})
+@RouteResolver({
+	deps: { params: ['id'] },
+	resolver: ({ route }) => Api.sendRequest('/web/dash/communities/games/' + route.params.id),
+})
+export default class RouteCommunitiesViewEditGames extends BaseRouteComponent {
+	@RouteStoreModule.State
+	community!: RouteStore['community'];
+
+	@RouteStoreModule.State
+	canLinkNewGames!: RouteStore['canLinkNewGames'];
+
+	@RouteStoreModule.State
+	linkGameCount!: RouteStore['linkGameCount'];
+
+	@RouteStoreModule.Mutation
+	setCanLinkNewGames!: RouteStore['setCanLinkNewGames'];
+
+	games: Game[] = [];
+
+	get hasLinkedGames() {
+		return this.games.length > 0;
+	}
+
+	get visibleGames() {
+		return this.games.filter(i => i.status === Game.STATUS_VISIBLE);
+	}
+
+	@Emit('games-change')
+	emitGamesChanged(_games: Game[]) {}
+
+	routeResolved($payload: any) {
+		// Request and populate games so hidden games are also shown.
+		this.games = Game.populate($payload.games);
+	}
+
+	isGameHidden(game: Game) {
+		return game.status !== Game.STATUS_VISIBLE;
+	}
+
+	async saveSort(sortedGames: Game[]) {
+		// Reorder the games to see the result of the ordering right away.
+		this.games!.splice(0, this.games!.length, ...sortedGames);
+
+		const sortedIds = sortedGames.map(i => i.id);
+		try {
+			const payload = await this.community.saveGameSort(sortedIds);
+			if (payload.success) {
+				this.emitGamesChanged(this.visibleGames);
+			}
+		} catch (e) {
+			console.error(e);
+			Growls.error(this.$gettext(`Could not save game arrangement.`));
+		}
+	}
```

Comment:
I think we should set the games on the community and call saveGameSort() before emitting.